File: java/src/main/java/com/android/frame/mvp/AATest/activity/weather/WeatherPresenter.java
package com.android.frame.mvp.AATest.activity.weather;

import com.android.frame.http.model.BaseResponse;
import com.android.frame.mvp.AATest.bean.WeatherBeanMvp;
import com.android.frame.mvp.BasePresenter;
import com.android.frame.mvp.extra.CustomObserver;

/**
 * Created by xuzhb on 2020/1/8
 * Desc:
 */
public class WeatherPresenter extends BasePresenter<WeatherView> {

    private WeatherView mView;
    private WeatherModel mModel;

    public WeatherPresenter(WeatherView view) {
        this.mView = view;
        mModel = new WeatherModel();
    }

    public void getWeatherInfo(String city) {
        mModel.getWeather(city)
                .subscribe(new CustomObserver<BaseResponse<WeatherBeanMvp>>(mView) {
                    @Override
                    public void onSuccess(BaseResponse<WeatherBeanMvp> response) {
                        mView.showData(response.getData());
                    }
                });
    }

}

File: kotlin/src/main/java/com/android/frame/mvp/AATest/activity/weather/WeatherPresenter.kt
package com.android.frame.mvp.AATest.activity.weather

import com.android.frame.http.model.BaseResponse
import com.android.frame.mvp.AATest.bean.WeatherBeanMvp
import com.android.frame.mvp.BasePresenter
import com.android.frame.mvp.extra.CustomObserver

/**
 * Created by xuzhb on 2020/1/4
 * Desc:
 */
class WeatherPresenter : BasePresenter<WeatherView>() {

    private val mModel by lazy { WeatherModel() }

    fun getWeatherInfo(city: String) {
        mView?.let {
            mModel.getWeather(city)
                .subscribe(object : CustomObserver<BaseResponse<WeatherBeanMvp>>(it) {
                    override fun onSuccess(response: BaseResponse<WeatherBeanMvp>) {
                        response.data?.let {
                            mView?.showData(it)
                        }
                    }
                })
        }
    }

}